### PR TITLE
Issue 52 mc

### LINKED ISSUE #52: Refactor DrugBank importer
- Currently uses a couple of Python modules -- should rewrite in Ruby for cleanliness purposes

### PR DESCRIPTION
**Pull request by mcannon068nw**
fixes #52 

Uses nokogiri SAX method to parse drugbank file, pull out relevant fields quickly as reading, and then create claims. These methods along with NCI importer can likely be abstracted and refactored to general XML parser methods

### CODE REVIEW

File: server/lib/genome/importers/file_importers/drugbank.rb [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    def run_parser
+      @drug_filter = DrugFilter.new
+      parser = Nokogiri::XML::SAX::Parser.new(drug_filter)
+      parser.parse(File.open(file_path))
+    end
+
+    def create_new_source
+      @source ||= Source.create(
+        {
+            base_url: 'https://go.drugbank.com/',
```
Comment (jsstevenson) [inline server/lib/genome/importers/file_importers/drugbank.rb:22]: base_url should be `https://go.drugbank.com/drugs/` to be consistent w/ other sources

File: server/lib/genome/importers/file_importers/drugbank.rb [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+            site_url: 'https://go.drugbank.com/',
+            citation: "DrugBank 5.0: a major update to the DrugBank database for 2018. Wishart DS, Feunang YD, Guo AC, Lo EJ, Marcu A, Grant JR, Sajed T, Johnson D, Li C, Sayeeda Z, Assempour N, Iynkkaran I, Liu Y, Maciejewski A, Gale N, Wilson A, Chin L, Cummings R, Le D, Pon A, Knox C, Wilson M. Nucleic Acids Res. 2017 Nov 8. doi 10.1093/nar/gkx1037. PubMed ID: 29126136",
+            source_db_version: '5.1.9',
+            source_db_name: 'DrugBank',
+            full_name: 'DrugBank - Open Data Drug & Drug Target Database',
+            license: '',
```
Comment (jsstevenson) [inline server/lib/genome/importers/file_importers/drugbank.rb:28]: Probably enough to just say "custom". Some other importers provide more of a description if you feel up for it.

File: server/lib/genome/importers/file_importers/drugbank.rb [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+            license_link: 'https://dev.drugbankplus.com/guides/drugbank/citing?_ga=2.29505343.1251048939.1591976592-781844916.1591645816'
+        }
+      )
+      @source.source_types << SourceType.find_by(type: 'interaction')
```
Comment (jsstevenson) [inline server/lib/genome/importers/file_importers/drugbank.rb:32]: I'm not even sure if I know what this is supposed to mean but it looks like DrugBank has previously also been tagged with the 'category' source type, too

File: server/lib/genome/importers/file_importers/drugbank.rb [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+                    create_interaction_claim_publication(interaction_claim, pmid)
+                end
+
+            end
+
+        end
+        backfill_publication_information
+    end
+
+
```
Comment (jsstevenson) [inline server/lib/genome/importers/file_importers/drugbank.rb:65]: Make sure to get these empty white space boys cleaned up

File: server/lib/genome/importers/file_importers/drugbank.rb [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+            end
+
+        when 'resource'
+            if @is_gene_identifier
+                @target_gene_id_resource_flag = true
+            end
+        end
+
+
+
```
Comment (jsstevenson) [inline server/lib/genome/importers/file_importers/drugbank.rb:157]: empty boys

File: server/lib/genome/importers/file_importers/drugbank.rb [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+    end
+
+    def characters(string)
+
+        if @drugbank_id_flag
+            @drugbank_id_flag = false
+            if is_not_salts
+                # p 'DRUGBANK ID: ' + string
```
Comment (jsstevenson) [inline server/lib/genome/importers/file_importers/drugbank.rb:165]: I would get rid of the commented `p` statements from here to line 198

File: server/lib/genome/importers/file_importers/drugbank.rb [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+            #append all sections to a record
+
+        when 'targets'
+        end
+
+    end
+
+    def end_document
+        p 'Finished parsing'
+    end
```
Comment (jsstevenson) [inline server/lib/genome/importers/file_importers/drugbank.rb:242]: I don't believe this method gets called?

File: server/lib/genome/importers/file_importers/drugbank.rb [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+
+        when 'external-identifiers'
+            @is_gene_identifier = false
+
+        when 'target'
+            @is_target = false
+            @record.append(current_drug_id,@current_drug_name,@current_target_name,@current_target_action,@current_target_pmid,@current_target_gene_identifiers,@current_target_gene_id_resources)
+            @all_records.append(@record)
+            @record = []
+            #append all sections to a record
```
Comment (jsstevenson) [inline server/lib/genome/importers/file_importers/drugbank.rb:233]: a little unclear what this note is referring to

**Review by jsstevenson — CHANGES_REQUESTED** [review]
👍 👍 👍 great work, just a few things to clean up

File: server/lib/genome/importers/file_importers/drugbank.rb (hunk 7 above)
Comment (jsstevenson) [reply server/lib/genome/importers/file_importers/drugbank.rb:242]: Oh, I'm guessing this is for Nokigiri? If so maybe just remove the puts statement so it finishes silently

File: server/lib/genome/importers/file_importers/drugbank.rb (hunk 8 above)
Comment (mcannon068nw) [reply server/lib/genome/importers/file_importers/drugbank.rb:233]: planning/debugging note, removed

**Review by jsstevenson — APPROVED** [review]

**Pull request merged**